operator/README.md

```diff
@@ -0,0 +1,275 @@
+- **Bring-your-own cluster support.** Supply an endpoint and the operator will skip provisioning and focus on federation/metadata tasks.
+- **Declarative database lifecycle.** Databases listed in the CRD are created if missing (including the heartbeat database when federation is enabled). Databases omitted from the spec are left untouched.
+- **Heartbeat-driven federation.** Partition clusters publish their discoverable schema and partition metadata to a hub. The hub turns that information into ADX tables and macro-expand functions that fan out across partitions.
+- **Status tracking and resumable reconciliation.** Conditions on the CRD capture progress (`Creating`, `Waiting`, `ClusterReady`, `Complete`, `PermissionRestricted`). Reconciliation resumes automatically after Azure operations finish.
+- **Safe interaction with partially managed resources.** 403 responses from Azure management APIs are tolerated: provisioning is skipped, but federation continues to operate.
+- **No implicit cleanup.** Deleting the CRD will not delete Azure assets; the operator intentionally leaves clusters and databases in place.
+
+## Prerequisites
+
+### Kubernetes
+
+- adx-mon operator installed in your cluster with the `ADXCluster` CRD registered.
+- The namespace where you create `ADXCluster` objects must allow the operator’s ServiceAccount to read/write those CRDs.
+
+### Azure access
+
+- The operator authenticates using `DefaultAzureCredential`. Provide a workload identity, pod-managed identity, service principal, or other supported credential chain that can:
+  - Register the `Microsoft.Kusto` provider.
+  - Create or update resource groups, ADX clusters, and databases when you request provisioning.
+  - Execute Kusto management commands ( `.show`, `.create table`, `.execute database script`, `.ingest inline` ) against every cluster endpoint referenced in your manifests.
+- For partition clusters, the identity used to write heartbeats must have ingestion rights on the federated cluster(s). The operator uses the `managedIdentityClientId` value from each federation target to request a user-assigned identity token when connecting to the hub.
+
+### Networking & endpoints
+
+- Every `endpoint` value must resolve from the operator’s pod network. HTTPS endpoints are expected and are treated as ADX data-plane URIs.
+- When zero-config provisioning is used, the resulting endpoint is written back into the CRD (`spec.endpoint`) once the cluster becomes reachable.
+
+## ADXCluster specification anatomy
+
+### Core cluster fields
+
+| Field | Required | Notes |
+| --- | --- | --- |
+| `spec.clusterName` | ✅ | Azure resource name for the ADX cluster. Must match `^[a-zA-Z0-9-]+$` and be ≤100 characters. |
+| `spec.endpoint` | ⛔ (optional) | Provide the HTTPS URI of an existing cluster to skip provisioning. |
```

Comment:
The red stop emojis are not the clearest - here it's intending to communicate that it's not required but usually red icons indicate to me that it is a deprecated field or something that should be avoided.  Inconsistent usage of `(optional)` as well. Maybe use the words Required and Optional?